File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        number=n
        sum=0
        prod=1
        while n>0:
            sum+= n%10
            prod*= n%10
            n= n//10
        total= sum+prod
        if number%total==0:
            return True
        return False

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.cpp
class Solution {
public:
    bool checkDivisibility(int n) {
        int number= n;
        int sum=0;
        int prod=1;
        while (number>0){
            sum= sum+(number%10);
            prod= prod*(number%10);
            number= number/10;
        }
        if (n%(sum+prod)==0) return true;
        return false;
    }
};